### PR TITLE
feat: Draft Drilldown POC

### PR DESCRIPTION
**Pull request by zhaoyongjie**
🏆 Enhancements
Draft Drilldown POC

### CODE REVIEW

File: packages/superset-ui-core/src/query/types/QueryFormData.ts [hunk 1]
```diff
@@ -133,6 +133,14 @@ export type ExtraFormDataOverride = ExtraFormDataOverrideRegular & ExtraFormData
 
 export type ExtraFormData = ExtraFormDataAppend & ExtraFormDataOverride;
 
+export type DrillDownType = {
+  hierarchy: string[];
+  currentIdx: number;
+  filters: QueryObjectFilterClause[];
+}
+
+export type OwnState = JsonObject & { drilldown: DrillDownType; }
```
Comment (villebro) [inline packages/superset-ui-core/src/query/types/QueryFormData.ts:142]: If we want to enforce type checking on `OwnState`, we probably need to introduce some named properties while moving the custom ones into nested ones. Something like
```ts
{
  drilldown: DrillDownType;
  customState: JsonObject;
}
```
 
In the above example the previous `ownState` would be moved into `customState`, while `drilldown` would be a reserved property for drilldown actions. However, that refactor should probably not be done in this PR, so I propose leaving `ownState` as `JsonObject` for now.

**Review by villebro — COMMENTED** [review]
One comment wrt typing, other than that looks good (pending the ongoing discussion on wether or not to have drilldown in chart metadata or dataset metadata).

**Comment by jhult** [comment]
@zhaoyongjie and @villebro,

Any news on this? It would be great to have this implemented.

**Comment by villebro** [comment]
The codebase on this repo has been moved to the main [Apache Superset repo](https://github.com/apache/superset), and consequently the repo is in the process of being archived. See the Superset Improvement Proposal for details: https://github.com/apache/superset/issues/13013 . While all currently open issues and PRs will be closed, we encourage you to reopen this PR on the main repo, which should be as simple as moving over any code changes as follows:
- core packages: `packages/**` (this repo) -> `superset-frontend/packages/**` (main repo)
- plugins: `plugins/**` (this repo) -> `superset-frontend/plugins/**` (main repo)

If you need help with the migration, please post a message on the SIP or reach out on the [community Slack](https://join.slack.com/t/apache-superset/shared_invite/zt-uxbh5g36-AISUtHbzOXcu0BIj7kgUaw).

**Pull request closed without merging**